executor.go

```diff
@@ -1,38 +1,43 @@
 package posthog
 
+import "sync"
+
 type executor struct {
-	queue chan func()
+	m         sync.Mutex
+	available int
+	closed    bool
 }
 
 func newExecutor(cap int) *executor {
 	e := &executor{
-		queue: make(chan func(), cap),
+		available: cap,
 	}
-	go e.loop()
 
 	return e
 }
 
 func (e *executor) do(task func()) bool {
-	select {
-	case e.queue <- task:
-		// task is enqueued successfully
-		return true
+	e.m.Lock()
+	defer e.m.Unlock()
 
-	default:
-		// buffer was full; inform the caller rather than blocking
```

Comment:
This buffer was never full, because in line 36 we were starting a new goroutine for each added task. The error here could happen only if the rate of starting goroutines would be lower than adding new tasks and I think this never happen to us.

Also, the test was incorrect.